### PR TITLE
Accept gzipped input throughout the pipeline

### LINKED ISSUE #76: Support gzip/xz-compressed input files throughout the pipeline
Raised by @Joon-Klaps reviewing #75 (https://github.com/nf-core/phyloplace/pull/75#discussion_r3869937102): should `--refseqfile` (and by extension other input files) support compressed input, e.g. `.gz`/`.xz`?

Currently `isFastaFile()` in `workflows/phyloplace.nf`, added in #75, reads the first line of `--refseqfile` directly via `path.withReader` to sniff whether it's FASTA (for the embedded-taxonomy fallback from #66). A gzip/xz-compressed file wouldn't be detected as FASTA this way -- it would silently be treated as "not FASTA" rather than erroring, since the raw compressed bytes don't start with `>`.

More broadly, several other pipeline inputs (`queryseqfile`, `refphylogeny`, `hmmfile`, `taxonomy`) aren't checked for compressed variants either, and whether/how each downstream tool (HMMER, Clustal Omega, EPA-NG, gappa, the new `custom/resolvetaxonomy` module) handles compressed input varies.

Worth scoping as its own piece of work rather than folding into #75: decide which inputs should support compression, verify each downstream tool's actual behavior with compressed input (don't assume), and add explicit decompression where a tool can't read it natively directly in the module that needs it (matching the pattern used elsewhere in this pipeline of decompressing inside the module rather than via a separate bridging module).

### PR DESCRIPTION
**Pull request by erikrikarddaniel**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/phyloplace/tree/master/docs/CONTRIBUTING.md)
- [x] If necessary, also make a PR on the nf-core/phyloplace _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core pipelines lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

## Description

Closes #76.

Every sequence, tree, profile and taxonomy file the pipeline reads may now be gzipped, whether it is named in a samplesheet or given as an individual parameter.
The samplesheets themselves stay uncompressed CSV.
Gzip is the only compression supported; any other format is rejected when parameters and samplesheets are validated.

### What each tool actually does with compressed input

Measured by running each tool against `.gz` input in its own container, rather than assumed:

| Tool | `.gz` |
| --- | --- |
| `hmmbuild`, `hmmsearch`, `hmmalign`, `esl-reformat`, `esl-alimask`, `hmmfetch` | works |
| `clustalo` (`-i`, `--profile1`) | works, output identical to plain input |
| **`mafft`** (positional input) | **writes zero bytes, exits 0, prints nothing** |
| `epa-ng` `--split`, `--ref-msa`, `--query` | works, output byte-identical to plain input |
| **`epa-ng --tree`** | **`Treeparsing failed! syntax error, unexpected STRING, expecting '('`** |
| `gappa --jplace-path` | works |
| **`gappa examine assign --taxon-file`** | **`A line in the taxon file didn't have two tab separated columns`** |
| `seqtk subseq`, `duckdb read_csv` | works |

So the pipeline decompresses for exactly three inputs; the rest of the chain reads gzip natively and needed nothing.

### Changes

Vendored modules, updated from nf-core/modules#12991:

- `mafft/align` — the positional input goes through `<(unpigz -cdf ...)`, the same idiom the module's own `--add*` options already use.
- `epang/place` — `--tree` is decompressed before the run.
- `gappa/examineassign` — same for `--taxon-file`.
- `custom/resolvetaxonomy` — opens its inputs through Python's `gzip`, sniffed on the magic bytes, and strips `.gz` before deriving the resolved sequences' extension, which otherwise named the output `*.resolved.gz`. The output declaration, the script and the stub now share one function for that instead of carrying three copies.

Pipeline:

- `isFastaFile()` reads through `GZIPInputStream`. Without it a gzipped `refseqfile` sniffs as "not FASTA" and silently skips taxonomy resolution rather than erroring.
- Every file parameter and samplesheet column has an extension pattern accepting the plain or `.gz` name, identical across the three schemas. Its error message lists the accepted extensions and says no other compression is supported.

### Gzipped intermediates, and a bug it uncovered

The Clustal Omega and MAFFT alignments are now written gzipped, like the rest of the pipeline's intermediates (`hmmsearch`, `hmmbuild`, `hmmalign`, `esl-reformat`, `esl-alimask`, `epang/split`, `epang/place` and `seqtk/subseq` already were).

Turning that on exposed a latent bug in `clustalo/align`.
Its `compress` branch wrote output as `-o >(pigz -cp N > file.gz)`, and the shell does not wait for the process inside `>(...)`, so the task could exit before pigz had flushed and leave a **truncated** `.gz`.
Every output above roughly 120 KB came out short; only the small ones won the race, which is why the minimal test profiles passed.
EPA-NG tolerates a short gzip stream by *hanging* on it rather than failing, so the symptom was a task sitting for the full one-hour `time` limit and dying with exit 143 — which reads like an executor problem, not a data problem.
Fixed upstream in nf-core/modules#13005, with the `&& wait $!` from #12894. That PR also adds the `compress_alignment` input to `fasta_newick_epang_gappa` that turns this on. The pipeline now carries no local patches. `mafft/align`'s `| pigz -cp` form is a real pipeline, so the shell waits, and it was verified unaffected.

### Tests

Two new profiles and pipeline tests, using fixtures added in nf-core/test-datasets#2277:

- `test_gzipped` — the mafft branch with gzipped query, reference, tree and taxonomy, covering the mafft, EPA-NG tree and gappa taxon-file paths at once.
- `test_gzipped_hmmfile` — the hmmer branch with a gzipped `hmmfile`, which the mafft branch never reaches.

`tests/compressed_input.nf.test` checks that a `.zip` input is rejected with that message, as a parameter and as a samplesheet column.

Both snapshots show the patches genuinely firing: `mafft/placement.fas.gz`, `epang/placement.epa_result.jplace.gz`, `gappa/placement.taxonomy.profile.tsv` with non-empty row counts, and `resolvetaxonomy/placement.resolved.alnfaa` / `.resolved.faa` with the `.gz` correctly stripped.

Full suite is 12/12 green. The existing snapshots change only in the `*.aln` → `*.aln.gz` and `*.fas` → `*.fas.gz` file names — every md5 is unchanged, since nft-utils hashes decompressed content, so the compression flip is provably content-neutral.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01Dpvh3T7aVgU5XyjqcYQLQD

### CODE REVIEW

**Comment by pinin4fjords** [comment]
Wouldn't it be nicer to PR nf-core/modules so everyone benefits from the ability to handle gzipped inputs, rather than doing all the patching?

**Comment by erikrikarddaniel** [comment]
> Wouldn't it be nicer to PR nf-core/modules so everyone benefits from the ability to handle gzipped inputs, rather than doing all the patching?

Very good point. It was definitely discussed between me and Claude -- I was lazy and won the argument -- and since there's no rush with the release, and there's now a majority for that, I'll go ahead.

File: subworkflows/nf-core/fasta_newick_epang_gappa/main.nf [hunk 1]
```diff
@@ -96,7 +96,7 @@ workflow FASTA_NEWICK_EPANG_GAPPA {
         [ ],
         ch_clustalo_data.map { it -> it.data.refseqfile },
         [ ],
-        false
+        true
```
Comment (pinin4fjords) [inline subworkflows/nf-core/fasta_newick_epang_gappa/main.nf]: You're patching this just to call the module differently? Sorry to be a broken record, but you should update the subworkflow interface's `take:` block to give you this control.

Patches should really be a last resort, I don't think it's called for here, other folks might want the same control you do.

File: modules/nf-core/clustalo/align/clustalo-align.diff [hunk 2]
```diff
@@ -0,0 +1,28 @@
+Changes in component 'nf-core/clustalo/align'
+'modules/nf-core/clustalo/align/meta.yml' is unchanged
+Changes in 'clustalo/align/main.nf':
+--- modules/nf-core/clustalo/align/main.nf
++++ modules/nf-core/clustalo/align/main.nf
```
Comment (pinin4fjords) [inline modules/nf-core/clustalo/align/clustalo-align.diff]: You might have to pro-actively pester about #12894 to get it resolved, or just PR yourself the more selective change without the script (I can approve). Better than leaving the patch

File: assets/schema_phyloplace_input.json [hunk 3]
```diff
@@ -51,7 +51,7 @@
             "taxonomy": {
                 "type": "string",
                 "format": "file-path",
-                "pattern": "^\\S+$",
+                "pattern": "^(\\S+\\.tsv(\\.gz)?|https?://\\S*/[^/.\\s]+)$",
                 "errorMessage": "Optional TSV file with taxonomy information."
             },
```
Comment (pinin4fjords) [inline assets/schema_phyloplace_input.json:56]: These new patterns are narrower than what's actually supported, verifiable within this PR's own diff:

- `taxonomy` (this field): only accepts `.tsv`/`.tsv.gz`, but `custom/resolvetaxonomy/meta.yml` — edited in this same PR — documents `.tax`/`.tax.gz` too. A `.tax` file is rejected here before it ever reaches the module that supports it.
- `refseqfile` (line 25): drops `alnfa`/`mfa`, both valid per this subworkflow's own pre-PR `meta.yml`.
- `queryseqfile` (line 19): only allows `fa|fna|faa|fasta`, but its own `errorMessage` still claims clustal/phylip/stockholm support.

Same three patterns are duplicated across this file, `schema_phylosearch_input.json`, and `nextflow_schema.json` — worth defining the accepted extensions once rather than drifting a fourth time.

File: subworkflows/local/utils_nfcore_phyloplace_pipeline/main.nf [hunk 4]
```diff
@@ -246,6 +246,21 @@ workflow PIPELINE_COMPLETION {
...
+//
+// Gzip is read natively by every tool in the chain, but HMMER's Easel cannot open xz at all,
+// so an xz input would otherwise fail several processes deep with a "misformatted file" error.
+//
+def rejectXz(label, value) {
+    if (value && value.toString().endsWith('.xz')) {
+        error("${label} is xz-compressed (${value}). Only gzip-compressed and uncompressed input are supported.")
+    }
+}
+
```
Comment (pinin4fjords) [inline subworkflows/local/utils_nfcore_phyloplace_pipeline/main.nf]: `.endsWith('.xz')` won't catch a URL with a trailing query string (`.../file.fa.xz?download=1`) — it'd pass this check and fail several steps later inside HMMER Easel with a confusing error instead of this function's clean one. Also unrelated to that: nothing tests this rejection path, positive or negative. Not blocking, but cheap to cover given it's the pipeline's clearest error message.

File: subworkflows/nf-core/fasta_newick_epang_gappa/main.nf (hunk 1 above)
Comment (erikrikarddaniel) [reply subworkflows/nf-core/fasta_newick_epang_gappa/main.nf]: Agreed. nf-core/modules#13005 (merged) adds a `compress_alignment` input to the subworkflow, plus the two log emits we were patching in. a01a734 takes it and drops the patch.

File: modules/nf-core/clustalo/align/clustalo-align.diff (hunk 2 above)
Comment (erikrikarddaniel) [reply modules/nf-core/clustalo/align/clustalo-align.diff]: Went with the selective change: just the `&& wait $!`, in the same modules PR (nf-core/modules#13005), with dnncha credited. Merged, and the patch is gone in a01a734, so the pipeline now carries no local patches.

File: assets/schema_phyloplace_input.json (hunk 3 above)
Comment (erikrikarddaniel) [reply assets/schema_phyloplace_input.json:56]: Fixed in cf2e277. `refseqfile` accepts `.alnfa`/`.mfa` and `taxonomy` `.tax`. `queryseqfile` stays FASTA-only because MAFFT needs FASTA, and its error message now says so.
The three schema files now carry identical patterns, generated from one list. I haven't found a way to share one definition between the parameter schema and the samplesheet schemas.

File: subworkflows/local/utils_nfcore_phyloplace_pipeline/main.nf (hunk 4 above)
Comment (erikrikarddaniel) [reply subworkflows/local/utils_nfcore_phyloplace_pipeline/main.nf]: Good catch, and it went further: the extension patterns only admit plain and `.gz` names, URL query strings included, so the xz check could never run.
I removed it and made the schema error messages say that no compression other than gzip is supported (4a5eba0). `tests/compressed_input.nf.test` checks that a `.zip` input is rejected with that message, as a parameter and as a samplesheet column.

**Review by pinin4fjords — APPROVED** [review]
All four flagged issues fixed and verified in code (not just trusted). Did a second pass on the rest of the diff: schema/meta.yml/modules.json/module SHAs/new test all check out. Two non-blocking nits I'm not holding this up for: the schema extension patterns are hand-kept-in-sync across three files rather than actually generated from one source, and usage.md doesn't mention non-gzip compression is rejected. CI green.

**Comment by erikrikarddaniel** [comment]
Thanks @pinin4fjords
